packages/firebase_crashlytics/firebase_crashlytics/android/src/main/java/io/flutter/plugins/firebase/crashlytics/FlutterFirebaseCrashlyticsPlugin.java

```diff
@@ -380,20 +386,47 @@ private boolean isCrashlyticsCollectionEnabled(FirebaseApp app) {
     SharedPreferences crashlyticsSharedPrefs =
         getCrashlyticsSharedPrefs(app.getApplicationContext());
 
-    if (crashlyticsSharedPrefs.contains("firebase_crashlytics_collection_enabled")) {
-      enabled = crashlyticsSharedPrefs.getBoolean("firebase_crashlytics_collection_enabled", true);
+    if (crashlyticsSharedPrefs.contains(FIREBASE_CRASHLYTICS_COLLECTION_ENABLED)) {
+      enabled = crashlyticsSharedPrefs.getBoolean(FIREBASE_CRASHLYTICS_COLLECTION_ENABLED, true);
     } else {
-      if (app.isDataCollectionDefaultEnabled()) {
+
+      Boolean manifestEnabled =
+          readCrashlyticsDataCollectionEnabledFromManifest(app.getApplicationContext());
+
+      if (manifestEnabled != null) {
+        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(manifestEnabled);
+        enabled = manifestEnabled;
+      } else {
         FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(true);
         enabled = true;
-      } else {
-        enabled = false;
       }
     }
 
     return enabled;
   }
 
+  private static Boolean readCrashlyticsDataCollectionEnabledFromManifest(
+      Context applicationContext) {
+    try {
+      final PackageManager packageManager = applicationContext.getPackageManager();
+      if (packageManager != null) {
+        final ApplicationInfo applicationInfo =
+            packageManager.getApplicationInfo(
+                applicationContext.getPackageName(), PackageManager.GET_META_DATA);
+        if (applicationInfo != null
+            && applicationInfo.metaData != null
+            && applicationInfo.metaData.containsKey(FIREBASE_CRASHLYTICS_COLLECTION_ENABLED)) {
+          return applicationInfo.metaData.getBoolean(FIREBASE_CRASHLYTICS_COLLECTION_ENABLED);
+        }
+      }
+    } catch (PackageManager.NameNotFoundException e) {
+      // This shouldn't happen since it's this app's package, but fall through to default
+      // if so.
+      Logger.getLogger().e("Could not read data collection permission from manifest", e);
+    }
+    return null;
```

Comment:
Should this default to false?

Reply:
I think it should default to true as stated in the [docs](https://firebase.google.com/docs/crashlytics/customize-crash-reports?platform=android#enable-reporting).

Reply:
Oh yes to true sorry, then you can simplify 
```
  if (manifestEnabled != null) {
        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(manifestEnabled);
        enabled = manifestEnabled;
      } else {
        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(true);
        enabled = true;
      }
```

to only 

```
        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(manifestEnabled);
        enabled = manifestEnabled;
```

Reply:
Right. That should work.